src/modules/access-token/rest-api/account-auth-middleware.ts

```diff
@@ -0,0 +1,29 @@
+import jsonwebtoken from 'jsonwebtoken';
+import { NextFunction, Request, Response } from 'express';
+import ConfigService from '../../config/config-service';
+import { UnAuthorizedAccessError } from '../types';
+
+export default class AccountAuthMiddleware {
+  public static ensureAccess(
+    req: Request,
+    _res: Response,
+    _next: NextFunction,
+  ): void {
+    // TODO: Implement this
+    // This will be a public method used by other services to ensure that the
+    // request has access to the said resource. It should decode the JWT token
+    // from the request and ensure the accountId associated with the token matches
+    // with the accountId in the request. If they don't, then it should pass
+    // UnAuthorizedAccessError in the next function
+    const authHeader = req.headers.authorization;
+    const token = authHeader.split(' ')[1];
```

Comment:
Would this throw if authorization header is missing in the request?

Reply:
Yes

Reply:
We should prevent it from throwing